compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -221,11 +232,26 @@ impl SelfProfilerRef {
     where
         A: Borrow<str> + Into<String>,
     {
-        let message = self
+        let message_and_format = self
             .print_verbose_generic_activities
-            .then(|| format!("{}({})", event_label, event_arg.borrow()));
+            .map(|format| (format!("{}({})", event_label, event_arg.borrow()), format));
 
-        VerboseTimingGuard::start(message, self.generic_activity_with_arg(event_label, event_arg))
+        VerboseTimingGuard::start(
+            message_and_format,
+            self.generic_activity_with_arg(event_label, event_arg),
+            false,
+        )
+    }
+
+    /// Like `verbose_generic_activity`, but `event_label` must be unique for a rustc session.
```

Comment:
More detail in this comment would be good. How is uniqueness ensured -- must the caller guarantee it? What happens if it's not unique?